File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        // sorted so can use two pointer
        let i = 0
        let j = numbers.length - 1
        while (i < j){
            let total = numbers[i] + numbers[j]
            if (total === target){
                return [i+1, j+1]
            } 
            else if (total > target) {
                j -= 1
            } else {
                i += 1
            }
        }

    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {} # key is the number and value is the indices
        for i in range(len(nums)):
            required = target - nums[i]
            if required in seen:

                return [seen[required], i]
            seen[nums[i]] = i

        